### PR TITLE
security: Enable authorize plugins to grant modify-only access

### PR DESCRIPTION
**Pull request by rhansen**

### CODE REVIEW

File: tests/backend/specs/socketio.js [hunk 1]
```diff
@@ -180,4 +186,54 @@ describe('socket.io access checks', () => {
...
+  });
+  it("level='modify' -> can modify", async () => {
+    const pad = await padManager.getPad('pad'); // Create the pad.
+    authorize = () => 'modify';
+    settings.requireAuthentication = true;
+    settings.requireAuthorization = true;
+    const res = await client.get('/p/pad').auth('user', 'user-password').expect(200);
+    socket = await connect(res);
+    const clientVars = await handshake(socket, 'pad');
+    assert.equal(clientVars.type, 'CLIENT_VARS');
```
Comment (JohnMcLear) [inline tests/backend/specs/socketio.js:217]: Is this a thorough enough test for modify?  Just getting ``CLIENT_VARS`` seems a little trusting that ``CLIENT_VARS`` means you can modify??

**Comment by rhansen** [comment]
Marked this PR as draft because I rebased this on top of #4346 so that will need to be merged first.

File: tests/backend/specs/socketio.js (hunk 1 above)
Comment (rhansen) [reply tests/backend/specs/socketio.js:258]: I added the following to the tests:

```
    assert.equal(clientVars.data.readonly, false);
```

**Comment by rhansen** [comment]
I rebased this onto the latest develop so it is ready to merge.

**Comment by JohnMcLear** [comment]
Should go in tonight, just wanna get testing passing first.

**Pull request merged**